Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -315,6 +319,26 @@ extension ActivityStreamPanel {
         let deleted = profile.prefs.arrayForKey(DefaultSuggestedSitesKey) as? [String] ?? []
         return suggested.filter({deleted.indexOf($0.url) == .None})
     }
+
+    func longPress(longPressGestureRecognizer: UILongPressGestureRecognizer) {
+        if longPressGestureRecognizer.state == UIGestureRecognizerState.Began {
+            let touchPoint = longPressGestureRecognizer.locationInView(self.view)
+            if let indexPath = tableView.indexPathForRowAtPoint(touchPoint) {
+                if indexPath.section == 1 {
+                    presentContextMenu(history[indexPath.row])
+                }
+            }
+        }
+    }
+
+    private func presentContextMenu(site: Site) {
+        let contextMenu = BlurTableViewController()
+        contextMenu.profile = profile
+        contextMenu.site = site
+        contextMenu.modalPresentationStyle = .OverCurrentContext
+        contextMenu.modalTransitionStyle = .CrossDissolve
+        self.presentViewController(contextMenu, animated: true, completion: nil)
```

Comment:
I notice that when I trigger the context menu in portrait then rotate to landscape, the background misses the top 10px of the screen. I have a feeling this is because of how we're presenting the view controller. We show the status bar initially but hide it on landscape which the view controller doesn't compensate for. Maybe try playing around with different presentation styles? We might eventually need to do a custom one if we want to do the blur effect fading in.